pkg/core/secrets/manager/manager.go

```diff
@@ -14,20 +14,22 @@ import (
 	secret_store "github.com/kumahq/kuma/pkg/core/secrets/store"
 )
 
-func NewSecretManager(secretStore secret_store.SecretStore, cipher secret_cipher.Cipher, validator SecretValidator) manager.ResourceManager {
+func NewSecretManager(secretStore secret_store.SecretStore, cipher secret_cipher.Cipher, validator SecretValidator, unsafeDelete bool) manager.ResourceManager {
```

Comment:
One thing I'd like to understand one day is why do we have so many resourceManager. Seems like an interceptor interface should be extracted from modification or maybe this validation should happen at a higher level (especially if we're make policies as plugins).

Reply:
I think the initial idea was to have a layer between storage an API for additional logic like
* extra validation (for example - does mesh exist?)
* setting ownership
* audits
* RBAC (was implemented on different layer after all...)

The unfortunate thing is that whenever something is applied on Kubernetes it does not go through it :/
I'm up for figuring out a better abstraction that works for both. I was thinking one day about relying more on events that we are emitting from storage, but "interceptors" might be a better idea.

Reply:
I like the idea of introducing interceptors

Reply:
https://github.com/kumahq/kuma/issues/4143